nano/node/node.cpp

```diff
@@ -2268,7 +2271,7 @@ void nano::node::process_confirmed (std::shared_ptr<nano::block> block_a, uint8_
 	auto hash (block_a->hash ());
 	if (ledger.block_exists (block_a->type (), hash))
 	{
-		add_confirmation_heights (hash);
+		confirmation_height_processor.add (hash);
```

Comment:
In the previous version, I believe this blocked until confirmation height was set. Now it schedules the block for processing. 

The confirmation callbacks may thus occur before the height is set. Is that a potential issue for callback receivers?  E.g using `block_info` to double check confirmation status.

During the initial confirmation height update (which may take a long time due to long chains), I think this non-blocking design means that calling `block_confirm` wont update height until the long chains are done with.

(Not sure if all this is just an upgrade documentation issue or not, but seems like a topic worth discussing as the block height updating necessarily takes a while.)

Reply:
I think this is a good observation, we want our confermation to be coherent from an external observer i.e. we shouldn't allow a case where a confirmation comes through yet the confirmation height hasn't been committed.

Reply:
`block_info` and related RPCs should return that the block is confirmed after a callback is done. It now checks the `confirmation_height_processor` to see if the block hash is being processed. This does not work iteratively (until the confirmation height is actually set), it will only return true for the top most block being confirmed.